test/e2e/build-sample-external-plugin.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# TODO: Dynamically set exteranl plugin destination based on OS platform
+EXTERNAL_PLUGIN_DESTINATION_PREFIX="${HOME}/Library/Application Support/kubebuilder/plugins"
+PLUGIN_NAME="sampleexternalplugin"
+PLUGIN_VERSION="v1"
+EXTERNAL_PLUGIN_DESTINATION="${EXTERNAL_PLUGIN_DESTINATION_PREFIX}/${PLUGIN_NAME}/${PLUGIN_VERSION}"
+EXTERNAL_PLUGIN_PATH="${EXTERNAL_PLUGIN_DESTINATION}/${PLUGIN_NAME}"
+
+if [ -d "$EXTERNAL_PLUGIN_DESTINATION" ]; then
+  echo "$EXTERNAL_PLUGIN_DESTINATION does exist."
+  if [ -e "$EXTERNAL_PLUGIN_PATH" ]; then
+    echo "clean up old binary..."
+    rm "$EXTERNAL_PLUGIN_PATH" 
+  fi
+else
+    mkdir -p "$EXTERNAL_PLUGIN_DESTINATION"
+fi
+
+SOURCE_DIR="./docs/book/src/simple-external-plugin-tutorial/testdata/sampleexternalplugin/v1"
+cd $SOURCE_DIR && go build -o $PLUGIN_NAME && mv $PLUGIN_NAME "$EXTERNAL_PLUGIN_PATH"
+
+kubebuilder init --plugins="${PLUGIN_NAME}/${PLUGIN_VERSION}" --help | grep --quiet "$PLUGIN_NAME"
```

Comment:
Nice touch, but this introduces a dependency on the `kubebuilder` binary being in your PATH when you run this script. Ideally the build script will not use something in the PATH but rather a location we can control during the build. There are a couple options here:
1. We remove this line and don't test that kubebuilder can initialize with this plugin as part of building the plugin
2. We build a local version of the kubebuilder binary from source to use for testing this as part of this script

Reply:
Looks like we already have a shell function for building kubebuilder here: https://github.com/kubernetes-sigs/kubebuilder/blob/804fed4839718539a6622bd04e910ec1b86f6be4/test/common.sh#L104